routers/api/v1/admin/user.go

```diff
@@ -200,7 +201,9 @@ func EditUser(ctx *context.APIContext) {
 	if form.FullName != nil {
 		u.FullName = *form.FullName
 	}
+	var emailChanged bool
 	if form.Email != nil {
+		emailChanged = !strings.EqualFold(u.Email, *form.Email)
```

Comment:
This and below `strings.EqualFold` doesn't seems safe for unicode strings.

```
func main() {
 s := string([]byte{0xE2, 0x84, 0xAA})
 log.Print(s)
 log.Print("K" == s)
 log.Print(strings.EqualFold("K", s))
}
2021/11/09 18:27:42 K
2021/11/09 18:27:42 false
2021/11/09 18:27:42 true
```

Maybe we need something like ` util.StringCompareIncase`

Reply:
I added `ValidateEmail` here, so I think it's that function's responsibility to remove all these invisible characters.